barman/cloud_providers/google_cloud_storage.py

```diff
@@ -333,3 +337,282 @@ def _delete_objects_batch(self, paths):
+        to snapshot_info.
+
+        The following additional field is saved to snapshots_info:
+
+            * gcp_project: The name of the project to which all snapshot-related
+                           resources belong
+
+        The following additional field is saved in the metadata dict for each snapshot:
+
+            * physical_block_size_bytes: The block size of the source disk in bytes.
+            * device_name: The name of the device associated with the source disk, as
+                           understood by the GCP API.
+            * size_gb: The size of the source disk in GB.
+
+        :param barman.infofile.LocalBackupInfo backup_info: Backup information.
+        :param str instance_name: The name of the VM instance to which the disks
+            to be backed up are attached.
+        :param str zone: The zone in which the snapshot disks and instance reside.
+        :param list[str] disks: A list containing the names of the source disks.
+        """
+        instance_metadata = self._get_instance_metadata(instance_name, zone)
+        snapshots = []
+        for disk_name in disks:
+            disk_metadata = self._get_disk_metadata(disk_name, zone)
+            metadata = {
+                "block_size": disk_metadata.physical_block_size_bytes,
+                "size": disk_metadata.size_gb,
+            }
+            # Check disk is attached and find device name
+            attached_disks = [
+                d
+                for d in instance_metadata.disks
+                if d.source == disk_metadata.self_link
+            ]
+            if len(attached_disks) == 0:
+                raise SnapshotBackupException(
+                    "Disk %s not attached to instance %s" % (disk_name, instance_name)
+                )
+            # We should always have exactly one attached disk matching the name
+            assert len(attached_disks) == 1
+
+            metadata["device_name"] = attached_disks[0].device_name
+            metadata["device_path"] = self.DEVICE_PREFIX + attached_disks[0].device_name
+            metadata["name"] = self.take_snapshot(backup_info, zone, disk_name)
+            snapshots.append(metadata)
+
+        # Add snapshot metadata to BackupInfo
+        backup_info.snapshots_info = {
+            "gcp_project": self.project,
+            "provider": "gcp",
+            "snapshots": snapshots,
+        }
+
+    def delete_snapshot(self, snapshot_name):
+        """
+        Delete the specified snapshot.
+
+        :param str snapshot_name: The name used to reference the snapshot within GCP.
+        """
+        try:
+            resp = self.client.delete(
+                {
+                    "project": self.project,
+                    "snapshot": snapshot_name,
+                }
+            )
+        except NotFound:
+            # If the snapshot cannot be found then deletion is considered successful
+            return
+        resp.result()
+        if resp.done():
```

Comment:
As with the comment on L458, we should be checking `resp.error_code` to determine if the request was successful and also checking and logging any `resp.warnings`.